gateway/operator.go

```diff
@@ -5,152 +5,36 @@ import (
 	"net/http"
 	"strconv"
 
+	"github.com/infobloxopen/atlas-app-toolkit/query"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/metadata"
-	"google.golang.org/grpc/status"
-
-	"github.com/grpc-ecosystem/grpc-gateway/runtime"
-
-	"github.com/infobloxopen/atlas-app-toolkit/query"
 )
 
 const (
 	filterQueryKey           = "_filter"
-	filterMetaKey            = "operator-filter"
 	sortQueryKey             = "_order_by"
-	sortMetaKey              = "operator-sort"
 	fieldsQueryKey           = "_fields"
-	fieldsMetaKey            = "operator-fields"
 	limitQueryKey            = "_limit"
-	limitMetaKey             = "operator-limit"
 	offsetQueryKey           = "_offset"
-	offsetMetaKey            = "operator-offset"
 	pageTokenQueryKey        = "_page_token"
-	pageTokenMetaKey         = "operator-page-token"
 	pageInfoSizeMetaKey      = "status-page-info-size"
 	pageInfoOffsetMetaKey    = "status-page-info-offset"
 	pageInfoPageTokenMetaKey = "status-page-info-page_token"
-	paginationContextKey     = "pagination-context"
+
+	QUERY_URL = "QUERY_URL"
 )
 
 // MetadataAnnotator is a function for passing metadata to a gRPC context
 // It must be mainly used as ServeMuxOption for gRPC Gateway 'ServeMux'
 // See: 'WithMetadata' option.
 //
-// MetadataAnnotator extracts values of query operators from incoming
-// HTTP testRequest accroding to REST API Syntax.
-// E.g:
-// - _order_by="name asc,age desc"
-// - _fields="name,age"
-// - _filter="name == 'John'"
-// - _limit=1000
-// - _offset=1001
-// - _page_token=QWxhZGRpbjpvcGVuIHNlc2FtZQ
+// MetadataAnnotator stores request URL in gRPC metadata from incoming HTTP testRequest
 func MetadataAnnotator(ctx context.Context, req *http.Request) metadata.MD {
-	vals := req.URL.Query()
 	mdmap := make(map[string]string)
-
-	if v := vals.Get(sortQueryKey); v != "" {
-		mdmap[runtime.MetadataPrefix+sortMetaKey] = v
-	}
-	if v := vals.Get(fieldsQueryKey); v != "" {
-		mdmap[runtime.MetadataPrefix+fieldsMetaKey] = v
-	}
-
-	if v := vals.Get(filterQueryKey); v != "" {
-		mdmap[runtime.MetadataPrefix+filterMetaKey] = v
-	}
-
-	if v := vals.Get(offsetQueryKey); v != "" {
-		mdmap[runtime.MetadataPrefix+offsetMetaKey] = v
-	}
-
-	if v := vals.Get(limitQueryKey); v != "" {
-		mdmap[runtime.MetadataPrefix+limitMetaKey] = v
-	}
-
-	if v := vals.Get(pageTokenQueryKey); v != "" {
-		mdmap[runtime.MetadataPrefix+pageTokenMetaKey] = v
-	}
-
+	mdmap[QUERY_URL] = req.URL.String()
```

Comment:
For gateway intercepter which parse collection ops, we need to have access to the URL parameters, so in MetadaAnnotator we just store it metadata